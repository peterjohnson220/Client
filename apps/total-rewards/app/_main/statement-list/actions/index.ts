export * from './statement-grid.actions';
export * from './statement-list.page.actions';
