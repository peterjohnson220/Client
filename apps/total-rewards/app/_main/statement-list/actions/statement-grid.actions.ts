import { Action } from '@ngrx/store';
import { GridDataResult } from '@progress/kendo-angular-grid';

export const LOAD_STATEMENTS = '[Total Rewards/Statement List/Statement Grid] Load Statements';
export const LOAD_STATEMENTS_SUCCESS = '[Total Rewards/Statements/Statement Grid] Load Statements Success';
export const LOAD_STATEMENTS_ERROR = '[Total Rewards/Statements/Statement Grid] Load Statements Error';
export const UPDATE_SEARCH_TERM = '[Total Rewards/Statements/Statement Grid] Update Search Term';

export const OPEN_ACTION_MENU = '[Total Rewards/Statements] Open Action Menu';
export const CLOSE_ACTION_MENU = '[Total Rewards/Statements] Close Action Menu';

export class LoadStatements implements Action {
  readonly type = LOAD_STATEMENTS;
}

export class LoadStatementsSuccess implements Action {
  readonly type = LOAD_STATEMENTS_SUCCESS;
  constructor(public payload: GridDataResult) { }
}

export class LoadStatementsError implements Action {
  readonly type = LOAD_STATEMENTS_ERROR;
}

export class UpdateSearchTerm implements Action {
  readonly type = UPDATE_SEARCH_TERM;
  constructor(public payload: string) { }
}

export class OpenActionMenu implements Action {
  readonly type = OPEN_ACTION_MENU;
  constructor(public payload: string) { }
}

export class CloseActionMenu implements Action {
  readonly type = CLOSE_ACTION_MENU;
}

export type Actions =
  LoadStatements |
  LoadStatementsSuccess |
  LoadStatementsError |
  UpdateSearchTerm |
  OpenActionMenu |
  CloseActionMenu;
