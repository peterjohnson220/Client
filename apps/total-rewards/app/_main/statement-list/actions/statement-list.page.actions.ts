import { Action } from '@ngrx/store';
import { GridDataResult } from '@progress/kendo-angular-grid';

export const LOAD_STATEMENTS = '[Total Rewards/Statements] Load Statements';
export const LOAD_STATEMENTS_SUCCESS = '[Total Rewards/Statements] Load Statements Success';
export const LOAD_STATEMENTS_ERROR = '[Total Rewards/Statements] Load Statements Error';

export const UPDATE_SEARCH_TERM = '[Total Rewards/Statements] Update Search Term';

export const OPEN_ACTION_MENU = '[Total Rewards/Statements] Open Action Menu';
export const CLOSE_ACTION_MENU = '[Total Rewards/Statements] Close Action Menu';

// statements
export class LoadStatements implements Action {
  readonly type = LOAD_STATEMENTS;
}

export class LoadStatementsSuccess implements Action {
  readonly type = LOAD_STATEMENTS_SUCCESS;
  constructor(public payload: GridDataResult) { }
}

export class LoadStatementsError implements Action {
  readonly type = LOAD_STATEMENTS_ERROR;
}

export class UpdateSearchTerm implements Action {
  readonly type = UPDATE_SEARCH_TERM;
  constructor(public payload: string) { }
}

export class OpenActionMenu implements Action {
  readonly type = OPEN_ACTION_MENU;
  constructor(public payload: string) { }
}

export class CloseActionMenu implements Action {
  readonly type = CLOSE_ACTION_MENU;
}

export type Actions =
  LoadStatements |
  LoadStatementsSuccess |
  LoadStatementsError |
  UpdateSearchTerm |
  OpenActionMenu |
  CloseActionMenu;
