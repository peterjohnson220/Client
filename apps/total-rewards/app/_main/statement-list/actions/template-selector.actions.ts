import { Action } from '@ngrx/store';

export const LOAD_TEMPLATES = '[Total Rewards/Statement List/Template Selector] Load Templates';
export const LOAD_TEMPLATES_SUCCESS = '[Total Rewards/Statement List/Template Selector] Load Templates Success';
export const LOAD_TEMPLATES_ERROR = '[Total Rewards/Statement List/Template Selector] Load Templates Error';

export class LoadTemplates implements Action {
  readonly type = LOAD_TEMPLATES;
}

export class LoadTemplatesSuccess implements Action {
  readonly type = LOAD_TEMPLATES_SUCCESS;
  constructor(public payload: any[]) {}
}

export class LoadTemplatesError implements Action {
  readonly type = LOAD_TEMPLATES_ERROR;
}

export type TemplateSelectorActions =
  LoadTemplates |
  LoadTemplatesSuccess |
  LoadTemplatesError;
