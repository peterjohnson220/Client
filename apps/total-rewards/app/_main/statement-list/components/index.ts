export * from './grid-action-menu';
