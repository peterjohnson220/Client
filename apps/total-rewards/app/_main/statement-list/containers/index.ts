export * from './statements-grid/statements-grid.component';
