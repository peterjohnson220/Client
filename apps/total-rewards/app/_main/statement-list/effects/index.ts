export * from './statement-grid.effects';
export * from './template-selector.effects';
