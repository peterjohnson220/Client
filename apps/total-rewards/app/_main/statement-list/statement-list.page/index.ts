export * from './statement-list.page';
