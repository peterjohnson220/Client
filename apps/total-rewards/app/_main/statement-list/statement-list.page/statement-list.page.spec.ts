import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { combineReducers, Store, StoreModule } from '@ngrx/store';
import { of } from 'rxjs';

import * as fromRootState from 'libs/state/state';

import * as fromTotalRewardsActions from '../actions/statement-list.page.actions';
import * as fromTotalRewardsReducer from '../reducers/statement-list.page.reducer';
import { StatementListPageComponent } from './statement-list.page';

describe('TotalRewardsPageComponent', () => {
  let component: StatementListPageComponent;
  let fixture: ComponentFixture<StatementListPageComponent>;
  let store: Store<fromRootState.State>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot({
          ...fromRootState.reducers,
          totalRewards: combineReducers(fromTotalRewardsReducer.reducer)
        }),
        ReactiveFormsModule],
      declarations: [ StatementListPageComponent ],
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    store = TestBed.get(Store);
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(StatementListPageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should show the grid component when statements are returned', () => {
    component.statements$ = of([{} as any]);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should show the create new banner when no statements are returned', () => {
    component.statements$ = of([]);
    component.loadingStatements$ = of(false);
    component.searchTerm$ = of(null);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
