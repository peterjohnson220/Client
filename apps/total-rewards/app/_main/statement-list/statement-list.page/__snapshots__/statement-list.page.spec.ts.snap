// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalRewardsPageComponent should show the create new banner when no statements are returned 1`] = `
<pf-total-rewards-statement-list-page
  focusedTab$={[Function Store]}
  statementsLoading$={[Function Observable]}
  statementsLoadingError$={[Function Observable]}
  statementsSearchTerm$={[Function Store]}
  statementsTotal$={[Function Observable]}
  store={[Function Store]}
  tabs={[Function NgbTabset]}
>
  <pf-page-with-title-bar>
    
    <fa-icon
      aria-hidden="true"
      class="mr-2"
    />
    Total Rewards 
    
    <pf-input-debounce
      class="mr-2"
    />
    
    <div
      class="d-flex flex-column h-100"
    >
      <div
        class="h-100"
      >
        <ngb-tabset
          ng-reflect-destroy-on-hide="false"
        >
          <ul
            class="nav nav-tabs justify-content-start"
            role="tablist"
          >
            
            <li
              class="nav-item"
            >
              <a
                aria-controls="Statements-panel"
                aria-disabled="false"
                aria-expanded="true"
                class="nav-link active"
                href=""
                id="Statements"
                role="tab"
              >
                 Statements
                
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                aria-controls="Templates-panel"
                aria-disabled="false"
                aria-expanded="false"
                class="nav-link"
                href=""
                id="Templates"
                role="tab"
              >
                 Templates
                
              </a>
            </li>
          </ul>
          <div
            class="tab-content"
          >
            
            
            <div
              aria-expanded="true"
              aria-labelledby="Statements"
              class="tab-pane active"
              id="Statements-panel"
              role="tabpanel"
            >
              
              
              <pf-create-new-statement-banner />
              
            </div>
            
            
            <div
              aria-expanded="false"
              aria-labelledby="Templates"
              class="tab-pane "
              id="Templates-panel"
              role="tabpanel"
            >
              
              <pf-total-rewards-template-card-selector />
            </div>
            
          </div>
        </ngb-tabset>
      </div>
    </div>
  </pf-page-with-title-bar>
</pf-total-rewards-statement-list-page>
`;

exports[`TotalRewardsPageComponent should show the grid component when statements are returned 1`] = `
<pf-total-rewards-statement-list-page
  focusedTab$={[Function Store]}
  statementsLoading$={[Function Store]}
  statementsLoadingError$={[Function Store]}
  statementsSearchTerm$={[Function Store]}
  statementsTotal$={[Function Observable]}
  store={[Function Store]}
  tabs={[Function NgbTabset]}
>
  <pf-page-with-title-bar>
    
    <fa-icon
      aria-hidden="true"
      class="mr-2"
    />
    Total Rewards 
    
    <pf-input-debounce
      class="mr-2"
    />
    
    <div
      class="d-flex flex-column h-100"
    >
      <div
        class="h-100"
      >
        <ngb-tabset
          ng-reflect-destroy-on-hide="false"
        >
          <ul
            class="nav nav-tabs justify-content-start"
            role="tablist"
          >
            
            <li
              class="nav-item"
            >
              <a
                aria-controls="Statements-panel"
                aria-disabled="false"
                aria-expanded="true"
                class="nav-link active"
                href=""
                id="Statements"
                role="tab"
              >
                 Statements
                
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                aria-controls="Templates-panel"
                aria-disabled="false"
                aria-expanded="false"
                class="nav-link"
                href=""
                id="Templates"
                role="tab"
              >
                 Templates
                
              </a>
            </li>
          </ul>
          <div
            class="tab-content"
          >
            
            
            <div
              aria-expanded="true"
              aria-labelledby="Statements"
              class="tab-pane active"
              id="Statements-panel"
              role="tabpanel"
            >
              
              
              <div
                class="h-100"
              >
                <pf-statements-grid />
              </div>
              
            </div>
            
            
            <div
              aria-expanded="false"
              aria-labelledby="Templates"
              class="tab-pane "
              id="Templates-panel"
              role="tabpanel"
            >
              
              <pf-total-rewards-template-card-selector />
            </div>
            
          </div>
        </ngb-tabset>
      </div>
    </div>
  </pf-page-with-title-bar>
</pf-total-rewards-statement-list-page>
`;
